docs/faq/basics/interactions.md

```diff
@@ -0,0 +1,104 @@
+[RespondAsync] and
+[DeferAsync] let the API know you have succesfully received the command. This is also called 'acknowledging' a command.
+DeferAsync will not send out a response, RespondAsync will.
+[FollowupAsync] follows up on succesful acknowledgement. 
+
+> [!WARNING]
+> If you have not acknowledged the command FollowupAsync will not work! the interaction has not been resonded to, so you cannot follow it up!
+
+[RespondAsync]: xref:Discord.IDiscordInteraction
+[DeferAsync]: xref:Discord.IDiscordInteraction
+[FollowUpAsync]: xref:Discord.IDiscordInteraction
+
+## Bad form Exception when I try to create my commands, why do I get this?
+
+Bad form exceptions are thrown if the slash, user or message command builder has invalid values.
+The following options could resolve your error.
+
+#### Is your command name lowercase?
+
+If your command name is not lowercase, it is not seen as a valid command entry.
+`Avatar` is invalid; `avatar` is valid.
+
+#### Are your values below or above the required amount? (This also applies to message components)
+
+Discord expects all values to be below maximum allowed.
+Going over this maximum amount of characters causes an exception.
+
+> [!NOTE]
+> All maximum and minimum value requirements can be found in the [Discord Developer Docs].
+> For components, structure documentation is found [here].
+
+[Discord Developer Docs]: https://discord.com/developers/docs/interactions/application-commands#application-commands
+[here]: https://discord.com/developers/docs/interactions/message-components#message-components
+
+#### Is your subcommand branching correct?
+
+Branching structure is covered properly here: xref:Guides.SlashCommands.SubCommand
+
+## My interaction commands are not showing up?
+
+If you registered your commands globally, it can take up to 1 hour for them to register.
+Did you register a guild command (should be instant), or waited more than an hour and still don't have them show up?
+
+- Try to check for any errors in the console, there is a good chance something might have been thrown.
+
+- Register your commands after the Ready event in the client. The client is not configured to register commands before this moment.
+
+- Check if no bad form exception is thrown; If so, refer to the above question.
+
+- Do you have the application commands scope checked when adding your bot to guilds?
+![Scope check](images/scope.png)
+
+## There are many options for creating commands, which do I use?
+
+```cs
+private async Task ReadyAsync()
+{
+  // pull your commands from some array, everyone has a different approach for this.
+  var commands = _builders.ToArray();
+
+  // write your list of commands globally in one go.
+  await _client.Rest.BulkOverwriteGlobalCommands(commands);
+
+  // write your array of commands to one guild in one go.
+  // You can do a foreach (... in _client.Guilds) approach to write to all guilds.
+  await _client.Rest.BulkOverwriteGuildCommands(commands, /* some guild */);
```

Comment:
Should we use the methods on the `SocketGuild` class here?

Reply:
@Cenngo ?